test/verify/check-storage-lvm2

```diff
@@ -340,6 +360,532 @@ class TestStorageLvm2(storagelib.StorageCase):
+            self.dialog_set_val("size", 10)
+            self.dialog_apply()
+            self.dialog_wait_close()
+            self.content_row_wait_in_col(row, 1, expected_name)
+
+        create(1, "linear", "lvol0")
+        create(2, "raid0", "lvol1")
+        create(3, "raid1", "lvol2")
+        create(4, "raid10", "lvol3")
+        create(5, "raid5", "lvol4")
+        create(6, "raid6", "lvol5")
+
+        def wait_msg(row, msg):
+            self.content_tab_wait_in_info(row, 1, "Physical volumes" if row == 1 else "Stripes", msg)
+
+        def wait_partial(row):
+            wait_msg(row, "This logical volume has lost some of its physical volumes and can no longer be used.")
+
+        def wait_degraded(row):
+            wait_msg(row, "This logical volume has lost some of its physical volumes but has not lost any data yet.")
+
+        def wait_maybe_partial(row):
+            wait_msg(row, "This logical volume has lost some of its physical volumes but might not have lost any data yet.")
+
+        self.force_remove_disk(disk1)
+
+        wait_partial(1)         # linear is broken now
+        wait_partial(2)         # striped as well
+        wait_degraded(3)        # mirror is fine
+        wait_degraded(4)        # striped mirror as well
+        wait_degraded(5)        # raid5 and ...
+        wait_degraded(6)        # ... raid6 are doing their job
+
+        self.force_remove_disk(disk2)
+
+        wait_degraded(3)        # mirror is still fine
+        wait_maybe_partial(4)   # striped mirror is not sure anymore
+        wait_partial(5)         # raid5 is now toast
+        wait_degraded(6)        # but raid6 still hangs on
+
+        self.force_remove_disk(disk3)
+
+        wait_degraded(3)        # mirror is _still_ fine
+        wait_partial(4)         # striped mirror has lost more than half and is kaputt
+        wait_partial(6)         # raid6 is finally toast as well
+
+    def testLvmOnLuks(self):
+        b = self.browser
+        m = self.machine
+
+        self.skip_if_no_layouts()
+
+        # Make sure that Cockpit gets the layout description right for
+        # encrypted physical volumes
+
+        self.login_and_go("/storage")
+
+        disk = self.add_loopback_disk()
+        b.wait_in_text("#others", disk)
+        m.execute(f"echo einszweidrei | cryptsetup luksFormat --pbkdf-memory 32768 {disk}")
+        m.execute(f"echo einszweidrei | cryptsetup luksOpen {disk} dm-test")
+        self.addCleanup(m.execute, "cryptsetup close dm-test || true")
+        m.execute("vgcreate vgroup0 /dev/mapper/dm-test")
+        self.addCleanup(m.execute, "vgremove --force vgroup0 2>/dev/null || true")
+        m.execute("lvcreate vgroup0 -n lvol0 -l100%FREE")
+
+        b.wait_in_text("#devices", "vgroup0")
+        b.click('#devices .sidepanel-row:contains("vgroup0")')
+
+        self.content_tab_wait_in_info(1, 1, "Physical volumes", bn(disk))
+
+        testlib.sit()
```

Comment:
Uhh this should not be there! How did the tests go green?

Reply:
Ouch! I guess stdin is /dev/null or something similar and sit() hits EOF and returns immediately.

Reply:
Removed